middleware/authentication.js

```diff
@@ -0,0 +1,64 @@
+const {User, Product} = require('../models')
+const jwt = require('jsonwebtoken')
+
+const adminAuthentication = (req, res, next)=>{
+    if(!req.headers.access_token){
+        throw {name: 'NOT_LOGIN'}
+    }
+    try{
+        const decoded = jwt.verify(req.headers.access_token, process.env.JWT_SECRET)
+        User.findOne({where:{
+            id: decoded.userId,
+            role: 'Admin'
+        }})
+        .then(result=>{
+            if(result){
+                req.user = {
+                    id: result.id,
+                    role: result.role
+                }
+                next() 
+            }else{
+                throw{name: 'USER_NOT_FOUND'}
```

Comment:
disini jangan user not found dul, kan kita di authentication, jadi bilang aja 'NOT_AUTHENTICATED'

Reply:
siap ka, ato ACCESS_DENIED ya? biar errHandle nya sama

Reply:
boleh